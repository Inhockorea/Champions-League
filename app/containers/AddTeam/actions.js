import {
  CHANGE_TEAMNAME,
  ADD_TEAMNAME,
  ADD_TEAMNAME_SUCCESS,
  ADD_TEAMNAME_ERROR,
  RESET_MESSAGES,
} from './constants';

/**
 * Changes the input field of the AddTeam form
 *
 * @param  {string} teamname The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_TEAMNAME, and the teamnname(string);
 */

export function changeTeamname(teamname) {
  return {
    type: CHANGE_TEAMNAME,
    teamname,
  };
}

/**
 * Adding teamname, triggers root saga
 *
 * @return {object} An action object with a type of ADD_TEAMNAME;
 */

export function addTeamname() {
  return {
    type: ADD_TEAMNAME,
  };
}

/**
 * Once the POST method in API call was successful, addTeamnameSuccess is triggered
 *
 * @param  {object} message The response you get from the post request.
 *
 * @return {object} An action object with a type of ADD_TEAMNAME_SUCCESS, and the message(object);
 */
export function addTeamnameSuccess(message) {
  return {
    type: ADD_TEAMNAME_SUCCESS,
    message,
  };
}

/**
 * If the POST method in API call was unsuccessful, addTeamnameError is triggered
 *
 * @param  {error} error.
 *
 * @return {error} An action object with a type of ADD_TEAMNAME_ERROR, and error;
 */
export function addTeamnameError(error) {
  return {
    type: ADD_TEAMNAME_ERROR,
    error,
  };
}

/**
 * Reseting alert messeages rendered after api calls;
 *
 * @return {object} An action object with a type of RESET_MESSAGES
 */
export function resetMessages() {
  return {
    type: RESET_MESSAGES,
  };
}
