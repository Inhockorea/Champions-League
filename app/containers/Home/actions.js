import {
  LOAD_TEAMNAMES,
  LOAD_TEAMNAMES_SUCCESS,
  LOAD_TEAMNAMES_ERROR,
} from './constants';

/**
 * Load the team names(strings), triggers root_saga
 *
 * @return {object} An action object with a type of LOAD_TEAMNAMES
 */
export function loadTeamNames() {
  return {
    type: LOAD_TEAMNAMES,
  };
}

/**
 * When api call in getTeamNames is successful;
 *
 * @param  {string} names loaded data (array of strings) from API
 *
 * @return {object} An action object with a type of LOAD_TEAMNAMES_SUCCESS and names: [...strings];
 */

export function loadTeamNamesSucess(names) {
  return {
    type: LOAD_TEAMNAMES_SUCCESS,
    names,
  };
}

/**
 * When api call in getTeamNames is not successful and catches an error
 *
 * @param  {object} error
 *
 * @return {object} An action object with a type of LOAD_TEAMNAMES_ERROR, and error;
 */

export function loadTeamNamesError(error) {
  return {
    type: LOAD_TEAMNAMES_ERROR,
    error,
  };
}
