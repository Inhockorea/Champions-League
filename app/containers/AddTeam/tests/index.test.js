/**
 * Test the AddTeam
 */

import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddTeam, mapDispatchToProps } from '../index';
import { changeTeamname, addTeamname } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddTeam />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddTeam
            teamname=""
            loading={false}
            error={false}
            onsubmitTeamname={jest.fn()}
            onchangeTeamname={jest.fn()}
            dispatchResetMessages={jest.fn()}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onsubmitTeamname if teamname is null', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddTeam
            teamname=""
            loading={{ message: 'fetching failed' }}
            onchangeTeamname={() => {}}
            onsubmitTeamname={submitSpy}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onchangeTeamname', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onchangeTeamname).toBeDefined();
      });

      it('should dispatch onchangeTeamname when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const teamname = 'Dortmund';
        result.onchangeTeamname({ target: { value: teamname } });
        expect(dispatch).toHaveBeenCalledWith(changeTeamname(teamname));
      });
    });

    describe('onsubmitTeamname', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onsubmitTeamname).toBeDefined();
      });

      it('should dispatch addTeamname when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onsubmitTeamname();
        expect(dispatch).toHaveBeenCalledWith(addTeamname());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onsubmitTeamname(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
