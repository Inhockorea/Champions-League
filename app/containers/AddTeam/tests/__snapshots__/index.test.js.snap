// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTeam /> should render and match the snapshot 1`] = `
<article>
  <div>
    <section
      class="Section-ket8oq-0 CenteredSection-lpvr5o-0 epDwMw"
    >
      <h2
        class="H2-egphza-0 HsCXl"
      >
        <span>
          Do you have what it takes to win the Champions League?
        </span>
      </h2>
      <p>
        <span>
          Your team name has to be between 1 and 20 characters (number & symbols allowed)
        </span>
      </p>
      <p>
        <span>
          Keep in mind that no dulpicate names are allowed!
        </span>
      </p>
    </section>
    <section
      class="Section-ket8oq-0 joDnBK"
    >
      <h2
        class="H2-egphza-0 HsCXl"
      >
        <span>
          Add Your Team!
        </span>
      </h2>
      <form
        class="Form-sc-131jk1n-0 dvEhEW"
      >
        <label
          for="teamname"
        >
          <input
            class="Input-sc-1u4rld3-0 hjznZH"
            id="teamname"
            placeholder="Insert your team name"
            type="text"
            value=""
          />
        </label>
        <button
          class="Button-sc-700fna-0 jLQkUW"
        >
          Add Team
        </button>
      </form>
      <section
        class="ErrorArea__Section-j2gckk-0 cpOjvw"
      >
        <section
          class="Success__Warning-sc-1hdr56j-0 llmbwq"
        >
           
          
           
        </section>
        <section
          class="Warning-sc-1grgbrk-0 hwgZwO"
        >
           
          
           
        </section>
      </section>
    </section>
  </div>
</article>
`;
